Add A/B test to move user step to last position of `onboarding` flow

andrewserong opened the pull request:
This is a 50/50 A/B test to test out moving the user / account step to the end of the onboarding flow. Currently users have to enter their account details before progressing through the flow. This change would move it to the end, similar to how the signup flow used to be.

***Some caveats:***

A Happychat error is thrown in the console on the `plans` step because `getHappychatAuth` fails since there is no user currently logged in. The impact to the user is that the inline help button (question mark in the bottom right hand corner) isn't there throughout the signup flow. Again, this might be okay for the purposes of the test, but if we permanently move the `user` step, we might want to look into a way to keep it for logged out users.

#### Changes proposed in this Pull Request

* Create A/B test `moveUserStepPosition` and conditionally swap the defaultFlowName from `onboarding` to `onboarding-user-last`.
* When the user is in the `last` group, also add an additional flow that is a duplicate of the `ecommerce-onboarding` flow, but with the `user` step in the last position. This allows switching between the onboarding and ecommerce flows while keeping the user step last, and avoiding the known issue with broken transitions between flows: https://github.com/Automattic/wp-calypso/issues/32506
* Update the Signup component to _not_ render the step if in the process of performing a redirect. This avoids a BSOD issue when the `user` step renders and calls `saveSignupStep`, potentially putting the `user` step in the wrong position in the progress state.
* Update the login and initial-state behaviour to _not_ remove persisted logged out state, and to copy it to the current state if there is no existing persisted state for a logged in user, but there _is_ persisted state for a logged out user. This will allow a user to complete the signup process, get to the last step, sign in, and resume their progress from the last step instead of having to enter everything in again. The intention is to effectively reinstate this bug fix from last year, but now in Redux: https://github.com/Automattic/wp-calypso/pull/25690
* Don't hide the login link in the footer of the `user` step when the step is in a position other than 0

#### Demo of logging in at the final step and being redirected back to the plans step

![move-user-step-last-ab-test-small](https://user-images.githubusercontent.com/14988353/61854181-e3e7f080-af00-11e9-8208-748bb9efe4c5.gif)

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Add the `no-force-sympathy` config flag and set it to `true`, either in your `development.json` file, or as an environment variable before you run `npm run start`. This will switch off the "Sympathy" feature in your local development environment, which randomly clears the state on page load 25% of the time. Switching this off will ensure that the login redirect will work correctly.

##### Test variant

* In the Calypso dev menu at the bottom right of the screen, set the `moveUserStepPosition` value to `last`
* From logged out, go to `/start/` and it should take you immediately to the `site-type` step. Complete the flow with blog / business / or professional site types. The final step should be the `user` step. After creating an account, it should immediately take you to the loading screen for creating a new site, and then on to the checklist.
* Repeat the steps above but select the `online store` site type, and this should take you to `/start/online-store-onboarding/domains`

##### Test variant with logging in

Note: the post login redirect will not resume progress to the previous step on a Private browser in Firefox due to it not persisting IndexedDB beyond a page reload / redirect. However, Chrome Incognito browser will work (and normal Firefox should work fine, too).

* In the Calypso dev menu at the bottom right of the screen, set the `moveUserStepPosition` value to `last`
* From logged out, go to `/start/` and it should take you immediately to the `site-type` step. Complete the flow with blog / business / or professional site types. The final step should be the `user` step.
* Log in to an existing account, and it should redirect you to the final step (plans).
* Select a plan (e.g. Start with a free site) and you should be redirected to the checklist page of the new site.
* Repeat the steps above but select the `online store` site type, and this should take you to `/start/online-store-onboarding/domains`.

##### Test control

* In the Calypso dev menu at the bottom right of the screen, set the `moveUserStepPosition` value to `first`
* From logged out, go to `/start/` and it should take you to the user step
* Log in, and from the `site-type` step, select `online-store`. This should take you to `/start/ecommerce-onboarding/domains`
* Completing signup should be as it was on master before applying this AB test

andrewserong commented on client/blocks/login/index.jsx:
```diff
@@ -135,10 +137,24 @@ class Login extends Component {
+					this.props.signupProgress.filter( step => step.stepName !== 'user' )
+				);
+				if ( lastStep ) {
+					setResumeAfterLogin( lastStep );
```
Note to self: this doesn't appear to be updating the state (or at least it doesn't end up in the persisted state), so I'll need to investigate this further.

andrewserong replied:
I was calling the action directly instead of it being dispatched via `this.props.setResumeAfterLogin` 🤦‍♂️

andrewserong commented:
I believe this is ready for review now. I still need to write a test to cover the changes in `initial-state.js`, but I'd be keen for feedback on the approach I've taken for this A/B test, and any concerns or issues in the changes to logged out and logged in state (copying over the logged out state to a freshly logged in user). There's also some overlap with this A/B test and the proposed A/B test in https://github.com/Automattic/wp-calypso/pull/34768

spen commented on client/blocks/login/index.jsx:
```diff
@@ -135,10 +137,24 @@ class Login extends Component {
+			} );
+		} else {
+			if ( isArray( this.props.signupProgress ) ) {
+				const lastStep = last(
```
No need to change this, just a tip: lodash has a `findLast` method which, in this case, would look something like this:

```js
const lastStep = findLast( this.props.signupProgress,  step => step.stepName !== 'user' );
```

andrewserong replied:
Updated!

spen commented on client/signup/main.jsx:
```diff
@@ -577,6 +574,7 @@ class Signup extends React.Component {
 	}
 
 	render() {
+		// Prevent rendering a step if in the middle of performing a redirect or resuming progress.
```
Thanks for adding this :)

spen commented on client/state/initial-state.js:
```diff
@@ -108,10 +108,18 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
 	try {
-		const storedState = await getStoredItem( reduxStateKey );
+		let loggedOutState,
+			storedState = await getStoredItem( reduxStateKey );
```
Tiny nitpick, but let's give each declaration a new line `let`:
```js
let loggedOutState;
let storedState = await getStoredItem( reduxStateKey );
```

andrewserong replied:
Updated!

spen commented on client/state/signup/progress/actions.js:
```diff
@@ -8,6 +8,7 @@ import { includes, isEmpty, reduce, snakeCase, toPairs } from 'lodash';
  */
 import {
 	SIGNUP_PROGRESS_SAVE_STEP,
+	SIGNUP_PROGRESS_SET_RESUME_AFTER_LOGIN,
```
A nitpick with the action naming here:
`SET` would ideally be last. If we were to have other similar actions, such as for `CLEAR`ing `the state item, we would want that to be next to each other alphabetically:

```js
// out of order:
const SIGNUP_PROGRESS_CLEAR_RESUME_AFTER_LOGIN = ...
const SIGNUP_PROGRESS_RESET = ...
const SIGNUP_PROGRESS_SET_RESUME_AFTER_LOGIN = ...

// in order:
const SIGNUP_PROGRESS_RESET = ...
const SIGNUP_PROGRESS_RESUME_AFTER_LOGIN_CLEAR = ...
const SIGNUP_PROGRESS_RESUME_AFTER_LOGIN_SET = ...
```

andrewserong replied:
Good call! I'll do some renaming.

andrewserong replied:
Updated!

spen commented on client/state/signup/progress/reducer.js:
```diff
@@ -74,6 +75,16 @@ function saveStep( state, { step } ) {
 
+function setResumeAfterLogin( state, { resumeStep } ) {
+	debug( `Setting resume after login for step ${ resumeStep.stepName }` );
+	return map( state, function( step ) {
```
I wonder, would something like this do the job? Or do we only want to set the `resumeStep` if it does already exist in the array? If that's not an issue we can avoid having to iterate over the collection.

```js
function setResumeAfterLogin( state, { resumeStep } ) {
  debug( `Setting resume after login for step ${ resumeStep.stepName }` );

  return {
    ...state,
    [ resumeStep.stepName ]: { 
      ...resumeStep, 
      status: 'in-progress' 
    },
  };
};
```

p-jackson replied:
`state` is an array so I think we unfortunately have to iterate through.

I see that other reducers in the file call `updateStep` to share this logic which might be an option here too. Its behaviour is slightly different though, so might not work for you.

andrewserong replied:
Yep, I was thinking the same re: having to iterate over it because it's an array. I initially decided not to reuse the `updateStep` function because the behaviour is a little bit different, but I think I'll take another look because it would make it a little cleaner.

andrewserong replied:
I've updated this to use the `updateStep` call, looks cleaner now!

spen replied:
> state is an array

Doh!

spen commented on config/development.json:
```diff
@@ -28,6 +28,7 @@
 	"rebrand_cities_prefix": "rebrandcitiesdevelopmentsite",
 	"livechat_support_locales": [ "en", "es", "pt-br" ],
 	"features": {
+		"no-force-sympathy": true,
```
I think this is checked in accidentally right?
I've rejected based on this only, the other comments aren't blockers at all :)

p-jackson replied:
This is listed in the PR description so seems intentional. But I'm guessing if it's needed in `development` it's probably needed in the other environments too?

andrewserong replied:
This is here intentionally — apologies, I should have been clearer on what's going on here! Related to Phil's comment in https://github.com/Automattic/wp-calypso/pull/34770#pullrequestreview-266887926, in `development` only, the persisted state is randomly cleared 25% of the time.

Here's where it's set randomly: https://github.com/Automattic/wp-calypso/blob/81162c9e9c8c2be8e78cac04e4c8c8735963434a/client/state/initial-state.js#L77

This is where it gets called: https://github.com/Automattic/wp-calypso/blob/81162c9e9c8c2be8e78cac04e4c8c8735963434a/client/state/initial-state.js#L232

So the change only needs to happen in the development environment to prevent the state from being cleared. Because this PR relies on persisted state existing in order to do the transfer from logged out to logged in state, I thought it could be confusing in the future if we continued to have the state randomly cleared on some page loads. For example, if a couple of months from now another developer is testing the login behaviour locally and wonders why sometimes the redirect doesn't work, but sometimes it does. By using the feature flag, people can still get the clearing behaviour back if they want it, it just wouldn't be the default.

What do you think?

ramonjd replied:
Would it be crazy (or even work) to check the abtest here?

```
	if ( config.isEnabled( 'no-force-sympathy' ) || 'last' === abtest('moveUserStepPosition') ) {
		return false;
	}
```

The AB isn't permanent anyway.

andrewserong replied:
Nice way to do it @ramonjd! I'll try it out...

andrewserong replied:
Works nicely, I've updated it and removed the line from `development.json`, so I'll resolve this part of this discussion.

p-jackson commented on client/blocks/signup-form/index.jsx:
```diff
@@ -810,18 +810,14 @@ class SignupForm extends Component {
 			<LoggedOutFormLinks>
 				<LoggedOutFormLinkItem href={ logInUrl }>
-					{ flowName === 'onboarding'
+					{ flowName === 'onboarding' || flowName === 'onboarding-user-last'
```
IMO the "Already have a WordPress.com account?" copy makes more sense to me when the user step is at the end.

When it says "Log in to create _a_ site for your existing account" I have a sinking feeling that it's going to forget my progress and throw out all my good work. I think "Log in to create _this_ site _with_ your existing account" is more accurate, but the generic copy works too I think.

andrewserong replied:
Good point! For the purposes of the A/B test, I can just remove my change here and it'll work fine.

ramonjd replied:
Concur. At least to me, the question variant is less of a mouthful. Fewer nouns, less to think about.

andrewserong replied:
I've updated this one.

p-jackson reviewed (commented):
My only other comment is that the changes to `initial-state.js` look very global, like they don't just impact the signup page but runs everywhere.
I think I read somewhere that calypso in development does something like randomly dumps the localStorage data to make sure it's not being relied upon. Will this impact the case where the user is logged in but has an empty `redux-state-${userId}` store?

p-jackson commented:
I'm a bit confused by the "logging in" behaviour of going back to the plans step. It seems a little confusing to have to answer the same question twice.
On the other hand, it's nice to see the plans page again in the correct currency (when logged out I see in AUD because of geo-ip I guess, and after login I see NZD).

Still, if the pick the free plan initially doesn't it feel a bit pushy to show them pricing plans again?

andrewserong commented:
Thanks for the review @p-jackson! I replied here about the state being cleared locally: https://github.com/Automattic/wp-calypso/pull/34770#discussion_r307567316

You're right, it is a pretty global change, so I'd be keen to hear any other ideas on either making it safer, or thinking through what some of the unintended side-effects could be?

For the redirect back to the final step, I was thinking a bit about the initial fix that @taggon did that also resumes back to the last step the user was working on: https://github.com/Automattic/wp-calypso/pull/25690 — I felt a bit cautious about trying to get the Signup component to fire off creating the new site after someone logs in. Because we're dealing with the page reloading, I thought what if you got up to the login step, then decided not to continue with that site creation, and then at a later time went and logged in to WordPress.com and then went to create a site, it might then automatically create the site you were previously working on, which might feel unexpected. At least by redirecting back to the final step, the user has to do one more interaction to confirm what they're trying to do...

Of course I might just be overthinking this, and maybe the more obvious thing of _just continue creating the site_ makes more sense 🤔

ramonjd commented on test/e2e/specs/wp-signup-spec.js:
```diff
@@ -81,6 +81,10 @@ before( async function() {
 	this.timeout( mochaTimeOut );
 
+	beforeEach( async function() {
+		return await overrideABTests.setOverriddenABTests( driver, 'moveUserStepPosition', 'first' );
```
I just found out that maybe we no longer need to set the default 

https://github.com/Automattic/wp-calypso/blob/master/client/lib/abtest/README.md#updating-ab-tests-in-the-e2e-tests

Unless you're switching between the two... ?

andrewserong replied:
Nice! I didn't realise it uses the default variation by ... _default_. I'll remove it :)

andrewserong replied:
Removed!

andrewserong commented:
Thanks for the feedback everyone! I've resolved our conversations so that I could keep track of the changes I've made, but _do_ feel free to unresolve any of the conversations if there are further changes you'd like me to make!

ramonjd commented:
This works well for me. Kudos for getting it off the ground and running. Will give it another spin once you think it's ready to roll.

p-jackson commented:
> I thought what if you got up to the login step, then decided not to continue with that site creation, and then at a later time went and logged in to WordPress.com and then went to create a site, it might then automatically create the site you were previously working on, which might feel unexpected.

> Of course I might just be overthinking this

Ok yeah, this is a really good point @andrewserong! Not overthinking it at all, I would have totally missed that. Seems that since we're relying on local storage across redirects then going back to the final step is safest.

This actually reminds me of #34156 where we used a cookie to persist data across page navigation becase "Redux state is not guaranteed to persist across page redirects" (quoted is taken out of context 😅). But then, I don't know what sort of data size limitations we have for storing site creation parameters in a cookie.

Maybe we should just AB test what we've got now since we know it's not final. We could maybe change the split from 50/50 to e.g. 95/5 if we were paranoid about users responding negatively to being "pestered" with the same question. If it's all in my head we could roll it out to more users?

andrewserong commented:
Whoops! Just did a bad rebase here, and some folks were automatically added to review... apologies for the pings! I'll fix this up...

a8ci18n commented on client/blocks/login/index.jsx:
```diff
@@ -153,21 +166,22 @@ class Login extends Component {
 		} = this.props;
 
-		let headerText = translate( 'Log in to your account' );
+		let headerText = translate( 'Log in to your account.' );
```
:ok: This change is safe, we'll keep the existing 29 translations.

a8ci18n commented on client/blocks/login/index.jsx:
```diff
@@ -153,21 +166,22 @@ class Login extends Component {
 		if ( isManualRenewalImmediateLoginAttempt ) {
-			headerText = translate( 'Log in to update your payment details and renew your subscription' );
+			headerText = translate(
+				'Log in to update your payment details and renew your subscription.'
```
:ok: This change is safe, we'll keep the existing 24 translations.

a8ci18n commented on client/blocks/login/index.jsx:
```diff
@@ -153,21 +166,22 @@ class Login extends Component {
 			headerText = translate( 'Two-Step Authentication' );
 		} else if ( socialConnect ) {
-			headerText = translate( 'Connect your %(service)s account', {
+			headerText = translate( 'Connect your %(service)s account.', {
```
:ok: This change is safe, we'll keep the existing 24 translations.

a8ci18n commented on client/blocks/login/index.jsx:
```diff
@@ -238,7 +252,7 @@ class Login extends Component {
 			);
 		} else if ( isJetpack ) {
-			headerText = translate( 'Log in to your WordPress.com account to set up Jetpack' );
+			headerText = translate( 'Log in to your WordPress.com account to set up Jetpack.' );
```
:ok: This change is safe, we'll keep the existing 24 translations.

andrewserong committed:
Add A/B test to move user step to last position of `onboarding` flow (#34770)

This commit manually squashes and rebases the following changes:

* Add A/B test to move user step to last position of `onboarding` flow

* Set `resumingStep` in main Signup class when redirecting the position

* Update login and initial-state to preserve logged out state

* Add test for setResumeAfterLogin reducer

* Prevent clearing state in local development environment if A/B test is active

andrewserong commented:
Manually rebased and squashed my commits, so the PR should be a little cleaner now.

andrewserong commented:
@jsnajdr I've just added you as a reviewer (hope that's okay!) for this PR. If you have time, I'd be keen to get your thoughts on the changes I'm proposing to `initial-state.js` to transfer state from a logged out session to a logged in session — this will enable us to have a user log in at the end of a signup flow, and resume their progress at the end of the signup instead of having to start from the beginning again. I'm aware I'm making a pretty global change here, and wanted to get some input from Calypso folks on the potential implications here. And I still need to add some unit tests for the change, of course. The main changes I'm referring to are here: https://github.com/Automattic/wp-calypso/pull/34770/files#diff-e18463a8ac59a4ed5a9b2f702e4cc4a3

I'm wrapping up for the week, but will be back at it on Monday working on this PR. Thanks!

jsnajdr commented on client/state/initial-state.js:
```diff
@@ -108,10 +109,18 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
+		let loggedOutState;
+		let storedState = await getStoredItem( reduxStateKey );
+		if ( reduxStateKey !== 'redux-state-logged-out' ) {
+			loggedOutState = await getStoredItem( 'redux-state-logged-out' );
```
One more thing: is it only the `state.signup` subtree that we want to transfer from logged-out to logged-in state? Then I would pick it from the logged-out state explicitly and throw the rest away. That's a precaution against spreading garbage data 🙂 

One more idea how to transfer only the `signup` state:
1. Wrap the `signup` reducer with `withStorageKey` wrapper. That will cause it to persist to a separate storage key, `redux-state-123:signup`, instead of the general `redux-state-123`.
2. Move the fallback-to-logged-out logic to the `loadReducerState` function that fetches all these sub-states from storage. Then we can enable the fallback only for `signup` key.

That's another way, slightly better, to limit the fallback only to the `signup` reducer.

andrewserong replied:
Thanks for guiding me here @jsnajdr, that's a much better idea!

jsnajdr reviewed (commented):
@andrewserong The initial state changes look solid: I posted a few suggestions how to improve the code, but the basic idea is :+1:

andrewserong committed:
Remove abtest from initial-state

andrewserong commented on client/state/initial-state.js:
```diff
@@ -69,7 +70,7 @@ function shouldAddSympathy() {
 
 	// If `no-force-sympathy` flag is enabled, never clear persistent state.
-	if ( config.isEnabled( 'no-force-sympathy' ) ) {
+	if ( config.isEnabled( 'no-force-sympathy' ) || 'last' === abtest( 'moveUserStepPosition' ) ) {
```
Thanks, that is very good context to know. I've removed the abtest from the module.

andrewserong committed:
Wrap signup reducer in withStorageKey to enable restoring just the signup state from logged out users

This moves the logic for restoring signup state from a logged out store to the `loadReducerState` function, and only fires on the `signup` storage key. For this to work, before `loadReducerState` is called, instead of returning null for an empty initialStoredState, this change creates an empty initial state first, otherwise the `initialReducer` call will error out.

andrewserong commented on client/state/initial-state.js:
```diff
@@ -108,10 +109,18 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
 		debug( 'fetched stored Redux state from persistent storage', storedState );
 
-		if ( storedState === null ) {
+		if ( isNil( storedState ) && loggedOutState ) {
```
Removed.

andrewserong commented on client/state/initial-state.js:
```diff
@@ -127,7 +136,11 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
-		if ( ! subkey && ! verifyStoredRootState( deserializedState ) ) {
+		if (
+			! subkey &&
+			get( deserializedState, [ 'currentUser', 'id' ], null ) !== null &&
```
Thanks! I've removed this additional check.

andrewserong commented on client/state/initial-state.js:
```diff
@@ -108,10 +109,18 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
-		if ( storedState === null ) {
+		if ( isNil( storedState ) && loggedOutState ) {
+			debug( 'stored Redux state not found, loading from loggedOutState instead' );
+			storedState = cloneDeep( loggedOutState );
```
Yes, that feels safe. I've removed the `cloneDeep` call.

andrewserong commented:
@jsnajdr I've made a few changes, thanks again for your suggestions, it's very helpful!

I've simplified the fallback-to-logged-out logic and moved it to the `loadReducerState` function as you suggested, and I think it makes it more readable too!

To get this to work, I've made a couple of other changes, too. I've added a third parameter to `getStateFromLocalStorage` called `forceLoggedOutUser` because that function needs to be aware of which redux state key that we're attempting to fetch from. A boolean felt like a simple way to do it, so that the calls to `getStateFromLocalStorage` in `loadReducerState` can be nice and simple — we only fetch from the logged out signup state if there wasn't anything returned from the fetch to the logged in signup state.

The other change that I needed to make was to set an empty initial state for `initialStoredState` instead of returning `null`, otherwise the `getInitialStoredState` function would return prematurely after logging in, before `loadReducerState` is able to get the data from the logged out store. This means that overall `getInitialState` will now return a default initial state if there is no persisted state, instead of returning `null`. I'm keen to hear your opinion on whether or not this is an okay change? I ended up needing to do this, because without an initial state, the `initialReducer` calls in `loadReducerState` will error out.

I've adjusted a couple of existing tests to reflect that change, and added in two additional tests for the new behaviour in `initial-state` for this PR. Thanks again!

jsnajdr commented on client/state/initial-state.js:
```diff
@@ -104,8 +105,13 @@ function verifyStateTimestamp( state ) {
+	if ( forceLoggedOutUser ) {
+		reduxStateKey = 'redux-state-logged-out' + ( subkey ? ':' + subkey : '' );
+	} else {
+		reduxStateKey = getReduxStateKey() + ( subkey ? ':' + subkey : '' );
```
It makes sense to add `forceLoggedOutUser` parameter to `getReduxStateKey` and then call it as:
```js
const reduxStateKey = getReduxStateKey( forceLoggedOutUser ) + ( subkey ? ':' + subkey : '' );
```

andrewserong replied:
Good idea, I've done this in my separate PR for the initial state changes: https://github.com/Automattic/wp-calypso/pull/34982

jsnajdr reviewed (commented):
Hi @andrewserong, I like the changes to the initial state code. It would make sense to create a separate PR for them and merge it separately. I often use this tactic to manage large changes like this PR.

andrewserong commented:
Thanks @jsnajdr, that's a good idea! I've split out the initial state changes into a separate PR: https://github.com/Automattic/wp-calypso/pull/34982 — I'll update and rebase this PR once that one lands.

spen commented:
Just noting that I rebased the branch with `master`, hence the force-push! 👍

andrewserong committed:
Rename `online-store-onboarding` to `ecommerce-store-onboarding` so that it is compatible with PBD

andrewserong commented:
Assuming the e2e tests pass, this PR is now ready for a final review.

ramonjd commented:
> Assuming the e2e tests pass, this PR is now ready for a final review.

The e2e fails at the importer step. Unlikely that this branch is the cause:

https://20789-102822243-gh.circle-artifacts.com/0/home/circleci/wp-calypso/wp-calypso/test/e2e/screenshots/videos/can-then-enter-a-valid-url-of-a-site-to-import-2019-08-05T18-02-58.mpg

ramonjd commented on client/signup/steps/site-type/index.jsx:
```diff
@@ -8,6 +8,7 @@ import { connect } from 'react-redux';
 /**
  * Internal dependencies
  */
+import { abtest } from 'lib/abtest';
```
`abtest` already has an import statement above...

ramonjd replied:
Fixed

ramonjd committed:
removed duped abtest import

ramonjd reviewed (approved):
I've walked through the test scenarios, created sites using a few site types.

_Tried_ to confuse the back button... seems to hold up.

Also stopped mid-flow and hard refreshed on `/start`, and was taken back to the first incomplete step.

Once CI is done, I'm willing to say that we can launch this. 

It's mid-week and, so long as we can monitor analytics (by `flowName` perhaps?) then we'll have enough eyes to make sure things are ok.

Great work @andrewserong

andrewserong committed:
Add check to renderImportButton in signup to not render when the moveUserStepPosition A/B test is using the 'last' variation
